test/policies/proxy/proxy.test.js

```diff
@@ -0,0 +1,174 @@
+const path = require('path');
+const fs = require('fs');
+const request = require('supertest');
+const assert = require('chai').assert;
+const https = require('https');
+const express = require('express');
+const expressApp = express();
+
+const config = require('../../../lib/config');
+const gateway = require('../../../lib/gateway');
+
+const originalGatewayConfig = config.gatewayConfig;
+const port = 5998;
+
+const serverKeyFile = path.join(__dirname, '../../fixtures/certs/server', 'server.key');
+const serverCertFile = path.join(__dirname, '../../fixtures/certs/server', 'server.crt');
+const incorrectClientKeyFile = path.join(__dirname, '../../fixtures', 'agent1-cert.key');
+const clientKeyFile = path.join(__dirname, '../../fixtures/certs/client', 'client.key');
+const clientCertFile = path.join(__dirname, '../../fixtures/certs/client', 'client.crt');
+const chainFile = path.join(__dirname, '../../fixtures/certs/chain', 'chain.pem');
+
+describe('proxy policy', () => {
+  let app, service;
+
+  before('start service', () => {
+    expressApp.all('*', function (req, res) {
+      res.status(200).json({ port: port }).end();
+    });
+
+    service = https.createServer({
+      key: fs.readFileSync(serverKeyFile),
+      cert: fs.readFileSync(serverCertFile),
+      ca: fs.readFileSync(chainFile),
+      requestCert: true,
+      rejectUnauthorized: true
+    }, expressApp);
+
+    service.listen(port);
+  });
+
+  after('clean up', () => {
+    config.gatewayConfig = originalGatewayConfig;
+    service.close();
+  });
+
+  describe('proxyOptions', () => {
+    describe('when proxy options are specified on the serviceEndpoint', () => {
+      before((done) => {
+        setupGateway({
+          target: {
+            keyFile: clientKeyFile,
+            certFile: clientCertFile,
+            caFile: chainFile
+          }
+        }).then(apps => {
+          app = apps.app;
+          done();
+        });
+      });
+
+      after(() => {
+        app.close();
+      });
+
+      it('passes options to proxy', (done) => {
+        request(app).get('/endpoint')
+          .expect(200)
+          .expect('Content-Type', /json/)
+          .end(function (error, res) {
+            if (error) return done(error);
+
+            assert.ok(res.body.port === port);
```

Comment:
Nip: You can use `assert.strictEqual(res.body.port, port)`

Reply:
I can see this piece of code is replicated multiple times. Could you define it somewhere above and reuse where possible?

Reply:
Going to remove this check, as it's not at all required. The status and content-type is more than enough to validate the test.